Exerscise_3/main.cpp

```diff
@@ -7,20 +7,21 @@
 #include "Exam.h"
 #include "Teacher.h"
 #include "Student.h"
+#include <memory>
 
 
 
 template<class T>
-void fill_up_container(std::istringstream &iss, std::vector<std::pair<int, IRecord*>> &container){
-    IRecord* inst = new T(iss);
+void fill_up_container(std::istringstream &iss, std::vector<std::pair<int, std::shared_ptr<IRecord>>> &container){
```

Comment:
Why std::shared_ptr?
Will you share ownership of that object?

Reply:
@DoYouEvenCpp I did it because I had an error when I used unique_ptr and when I tried to emlace_back an instance to the container. But I founded that I need to use std::move this pointer to be able to add it to the container, that I can't make copies of a unique_ptr.

Reply:
Yes, passing of std::unique_ptr is possibly usually with std::move